### ISSUE TITLE
Differential eigensystems

### ISSUE THREAD

**Issue by seadra** [SciML/DifferentialEquations.jl#650]
(This is a feature request, apologies if this is the wrong place to post this.)

It'd be very nice to have the means to solve differential eigensystems, i.e., L f = λ f where L is a (partial) differential (matrix) operator and f is an eigenfunction with the eigenvalue λ.

Mathematica version 11 added support for solving differential eigensystems via [NDEigensystem](https://reference.wolfram.com/language/ref/NDEigensystem.html) (Stephen Wolfram briefly mentions about it in [his blog post announcing v11](https://blog.wolfram.com/2016/08/08/today-we-launch-version-11/) as well).

Such equations are very common in physics (Schrodinger equation, acoustics, structural mechanics, etc etc). I'm not sure whether the infrastructure for supporting this in mostly in place in Julia or not, but it would be a very strong and attractive feature to have in Julia ecosystem.

**Comment by ChrisRackauckas** [comment]
Yeah, differential eigensystems are something that would be nice to tackle somewhere in the SciML universe. They just haven't intersected with my research yet, which is why there isn't a package there. IIUC a lot of the methods for this kind of problem are more related to time-independent PDEs, which hasn't really intersected too much which DifferentialEquations.jl which has had a very strong focus on timestepping methods. To keep the scope of problems managable, we could probably tackle these by having problem types that output `LinearProblem` and `NonlinearProblem`, and then it would plug into our growing PDE ecosystem and rely on the rest of Julia to write linear solvers (and pull from PETSc, Trillinos, etc. that already do this part quite well).

So indeed it's good to have on the radar but probably won't be a focus or a strength for some time.
